### PR TITLE
add `nerdctl run --cidfile <FILE>`

### PR DESCRIPTION
**Pull request by smowafy**
First flag in the list in https://github.com/containerd/nerdctl/issues/215

One thing I'm not sure about is the category of this flag

### CODE REVIEW

File: run_test.go [hunk 1]
```diff
@@ -137,3 +137,18 @@ func TestRunExitCode(t *testing.T) {
 	assert.Equal(base.T, "exited", inspect123.State.Status)
 	assert.Equal(base.T, 123, inspect123.State.ExitCode)
 }
+
+func TestRunCIDFile(t *testing.T) {
```
Comment (AkihiroSuda) [inline run_test.go:141]:
```
=== RUN   TestRunCIDFile
    run_test.go:145: assertion failed: expression is false: res.ExitCode == exitCode: time="2021-05-24T17:38:22Z" level=fatal msg="image name needs to be specified"
        
--- FAIL: TestRunCIDFile (0.02s)
```

File: run.go [hunk 2]
```diff
@@ -734,3 +746,18 @@ func propagateContainerdLabelsToOCIAnnotations() oci.SpecOpts {
...
 }
+
+func writeCIDFile(path, id string) error {
+	if _, err := os.Stat(path); err == nil {
+		return errors.Errorf("Container ID file found, make sure the other container isn't running or delete %s", path)
+	}
+	f, err := os.Create(path)
+	defer f.Close()
+	if err != nil {
+		return errors.Errorf("Failed to create the container ID file: %s", err)
```
Comment (AkihiroSuda) [inline run.go:757]: nit: error string should start with a lower character in Go

File: run.go [hunk 3]
```diff
@@ -257,6 +261,14 @@ func runAction(clicontext *cli.Context) error {
 		id    = idgen.GenerateID()
 	)
 
+	cidfile := clicontext.String("cidfile")
+
+	if cidfile != "" {
```
Comment (AkihiroSuda) [inline run.go:266]: `if cidfile := clicontext.String("cidfile"); cidfile != "" {`

File: run.go [hunk 4]
```diff
@@ -227,6 +227,10 @@ var runCommand = &cli.Command{
 			Name:  "label-file",
 			Usage: "Read in a line delimited file of labels",
 		},
+		&cli.StringFlag{
+			Name:  "cidfile",
+			Usage: "Write the container ID to the file",
+		},
```
Comment (AkihiroSuda) [inline run.go:233]: Could you also update README.md to include `cidfile`

File: run.go [hunk 5]
```diff
@@ -734,3 +744,18 @@ func propagateContainerdLabelsToOCIAnnotations() oci.SpecOpts {
 		return oci.WithAnnotations(c.Labels)(ctx, oc, c, s)
 	}
 }
+
+func writeCIDFile(path, id string) error {
+	if _, err := os.Stat(path); err == nil {
+		return errors.Errorf("container ID file found, make sure the other container isn't running or delete %s", path)
+	}
+	f, err := os.Create(path)
```
Comment (fahedouch) [inline run.go:752]: I think you should do smthing like this to handles err différent from `isNotExist`

```
if _, err := os.Stat("/path/to/whatever"); err == nil {
  // throw erro
} else if os.IsNotExist(err) {
  // your logic here

} else {
  // handle other erro
}
```

File: run.go (hunk 2 above)
Comment (smowafy) [reply run.go]: The error message was inspired from https://github.com/docker/cli/blob/master/cli/command/container/create.go#L180, changed it to lowercase though

File: run.go (hunk 4 above)
Comment (smowafy) [reply run.go:233]: To which category though?

Comment (AkihiroSuda) [reply run.go:233]: `Metadata flags:` would be fine, I think

File: run.go (hunk 5 above)
Comment (smowafy) [reply run.go:754]: I added returning other errors, please check if this way makes sense. Also I used `errors.Is` according to https://golang.org/pkg/os/#IsExist

File: run_test.go (hunk 1 above)
Comment (smowafy) [reply run_test.go:141]: Replaced by just asserting failure if the file exists. Is that enough or should it be more specific?

File: run.go (hunk 4 above)
Comment (smowafy) [reply run.go:233]: Added

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**